optaplanner-quickstart/src/main/java/org/acme/optaplanner/domain/Lesson.java

```diff
@@ -0,0 +1,87 @@
+package org.acme.optaplanner.domain;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
+import org.optaplanner.core.api.domain.entity.PlanningEntity;
+import org.optaplanner.core.api.domain.lookup.PlanningId;
+import org.optaplanner.core.api.domain.variable.PlanningVariable;
+
+@PlanningEntity
+@Entity
+public class Lesson extends PanacheEntityBase {
+
+    @PlanningId
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @NotNull
+    private Long id;
+
+    @NotBlank
+    private String subject;
+    @NotBlank
+    private String teacher;
+    @NotBlank
+    private String studentGroup;
+
+    @PlanningVariable(valueRangeProviderRefs = "timeslotRange")
+    @ManyToOne
+    private Timeslot timeslot;
+    @PlanningVariable(valueRangeProviderRefs = "roomRange")
+    @ManyToOne
+    private Room room;
+
+    public Lesson() {
+    }
+
+    public Lesson(String subject, String teacher, String studentGroup) {
+        this.subject = subject.trim();
+        this.teacher = teacher.trim();
+        this.studentGroup = studentGroup.trim();
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public String getSubject() {
+        return subject;
+    }
+
+    public String getTeacher() {
+        return teacher;
+    }
+
+    public String getStudentGroup() {
+        return studentGroup;
+    }
+
+    public Timeslot getTimeslot() {
+        return timeslot;
+    }
+
+    public void setTimeslot(Timeslot timeslot) {
+        this.timeslot = timeslot;
+    }
+
+    public Room getRoom() {
+        return room;
+    }
+
+    public void setRoom(Room room) {
+        this.room = room;
+    }
+
+    @Override
+    public String toString() {
+        return subject + "(" + id + ")";
+    }
+
```

Comment:
No `equals` and `hashCode` on data beans is risky :)

Reply:
Why? :)

I'd like to keep this example as simple as possible, while still visually showing the full power of OptaPlanner on Quarkus. There is no reliance anywhere on equals/hashcode of those domain objects in this example.

Reply:
I'd like to code the entire example from scratch in 20 minutes. So anything that isn't really needed, I 'd like to avoid. I hope that makes sense :)

Reply:
I accepted the PR anyway, IMO, a lot of bugs happens because of data with no equals/hashCode, data bean often get used in Set, or Maps or simply for comparison in tests. If know that quickstarts are supposed to be as clean as possible, and again maybe another OCD of mine 😂